### DISCUSSION TITLE
Openssl@3.0 instead of 3.4 version for PHP

### DISCUSSION THREAD

**Discussion by MikKuba** [Homebrew/discussions#5822] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.12-60-g1d471a2
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1d471a2a194ffec63c6cf02cba7e37990557f658
Last commit: 21 hours ago
Branch: master
Core tap JSON: 21 Dec 16:45 UTC
Core cask tap JSON: 21 Dec 16:45 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.6 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 16.0.0 build 1600
Git: 2.47.1 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.6.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 16.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  imap-uw
  php@5.6
  php@7.0
  php@7.1
  php@7.3
  php@7.4
  php@8.0

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected header files:
  /usr/local/include/node/cppgc/allocation.h
  /usr/local/include/node/cppgc/common.h
 ...and hundreds another files...

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  mariadb
  php
  openssl@3

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo openssl/old) checkout 
  git -C $(brew --repo openssl/legacy) checkout
```

### Description of issue

Hi,

I need to reinstall my php (at least php@8.0) with openssl@3.0.

I have set openssl to version 3.0:
` % openssl version`
`OpenSSL 3.0.15 3 Sep 2024 (Library: OpenSSL 3.0.15 3 Sep 2024)`

But the PHP still using version 3.4:

`% php -i | grep "OpenSSL"  `
`SSL Version => OpenSSL/3.4.0 (SecureTransport)`
`OpenSSL support => enabled`
`OpenSSL Library Version => OpenSSL 3.4.0 22 Oct 2024`
`OpenSSL Header Version => OpenSSL 3.4.0 22 Oct 2024`
`Native OpenSSL support => enabled`

The goal is to set openssl@3.0 also for PHP.
I've tried to reinstall PHP and few tips from ChatGPT, but without success. 

Thanks for any hints!

**Comment by p-linnane** [comment]
We build formulae with the latest version of dependents wherever possible. You will need to manually edit the `php` formula, change the dependency to `openssl@3.0` and recompile `php` from source. We do not officially support this, so you are on your own if you decide to try it.

**Reply by MikKuba** [reply]
> We build formulae with the latest version of dependents wherever possible. You will need to manually edit the `php` formula, change the dependency to `openssl@3.0` and recompile `php` from source. We do not officially support this, so you are on your own if you decide to try it.

@p-linnane I've tried to change formulae and edit lines to:
`depends_on "openssl@3.0"`

`# Use OpenSSL cert bundle`
`   openssl = Formula["openssl@3.0"]`
`   %w[development production].each do |mode|`
`      inreplace "php.ini-#{mode}", /; ?openssl\.cafile=/,`
`       "openssl.cafile = \"#{openssl.pkgetc}/cert.pem\""`
`      inreplace "php.ini-#{mode}", /; ?openssl\.capath=/,`
`        "openssl.capath = \"#{openssl.pkgetc}/certs\""`
`    end`

After running `brew reinstall php@8.0 --build-from-source`, `brew services restart php@8.0` and also `brew cleanup`, I still see openssl 3.4

This is message after reinstall php@8.0 without any warnings or errors:




To enable PHP in Apache add the following to httpd.conf and restart Apache:
    LoadModule php_module /opt/homebrew/opt/php@8.0/lib/httpd/modules/libphp.so

    <FilesMatch \.php$>
        SetHandler application/x-httpd-php
    </FilesMatch>

Finally, check DirectoryIndex includes index.php
    DirectoryIndex index.php index.html

The php.ini and php-fpm.ini file can be found in:
    /opt/homebrew/etc/php/8.0/

php@8.0 is keg-only, which means it was not symlinked into /opt/homebrew,
because this is an alternate version of another formula.

If you need to have php@8.0 first in your PATH, run:
  echo 'export PATH="/opt/homebrew/opt/php@8.0/bin:$PATH"' >> ~/.zshrc
  echo 'export PATH="/opt/homebrew/opt/php@8.0/sbin:$PATH"' >> ~/.zshrc

For compilers to find php@8.0 you may need to set:
  export LDFLAGS="-L/opt/homebrew/opt/php@8.0/lib"
  export CPPFLAGS="-I/opt/homebrew/opt/php@8.0/include"

To restart shivammathur/php/php@8.0 after an upgrade:
  brew services restart shivammathur/php/php@8.0
Or, if you don't want/need a background service you can just run:
  /opt/homebrew/opt/php@8.0/sbin/php-fpm --nodaemonize
==> Summary
🍺  /opt/homebrew/Cellar/php@8.0/8.0.30_5: 496 files, 77.7MB, built in 2 minutes 54 seconds
==> Running `brew cleanup php@8.0`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
Removing: /Users/michalkuba/Library/Caches/Homebrew/php@8.0--8.0.30.tar.gz... (15MB)






Is there any other option, how to get openssl@3.0 for my PHP?
Thanks!

**Reply by carlocab** [reply]
> Is there any other option, how to get openssl@3.0 for my PHP?

You probably did the `depends_on` replacement wrong, which is why it didn't work.

This might have to be something you need to work out on your own. Good luck!

**Reply by gromgit** [reply]
> I still see openssl 3.4

If you're checking the OpenSSL version by running `php -i | grep "OpenSSL"` as you did previously, note that you're almost certainly still running the Homebrew mainline PHP. You can check this by simply running `php --version`.

As the `brew install` message you posted reads:

> php@8.0 is keg-only, which means it was not symlinked into /opt/homebrew, because this is an alternate version of another formula.

You should instead be testing with `/opt/homebrew/opt/php@8.0/bin/php -i | grep OpenSSL`.

**Comment by SMillerDev** [comment]
> I need to reinstall my php (at least php@8.0) with openssl@3.0.

That version of PHP is no longer supported, so it won't be enabled in Homebrew/core, in addition to https://github.com/orgs/Homebrew/discussions/5822#discussioncomment-11638810
